lib/veres-one-client.js

```diff
@@ -0,0 +1,187 @@
+/*!
+ * Copyright (c) 2018 Veres One Project. All rights reserved.
+ */
+'use strict';
+
+const axios = require('axios');
+const https = require('https');
+const WebLedgerClient = require('web-ledger-client');
+
+const DEFAULT_MODE = 'test';
+
+class VeresOneClient {
+  /**
+   * @param injector {Injector}
+   *
+   * @param hostname {string} Hostname of the ledger (points to a load balancer
+   *   or a specific node).
+   * @param ledger {WebLedgerClient}
+
+   * @param [mode='test'] {string} One of 'dev'/'test'/'live'. Determines https
+   *   agent settings (does not reject unsigned certs in 'dev' mode, for
+   *   example).
+   * @param [logger]
+   * @param [httpsAgent] {Agent}
+   */
+  constructor({injector, hostname, ledger, mode, logger, httpsAgent}) {
+    this.injector = injector;
+    this.hostname = hostname;
+    if(!hostname) {
+      throw new Error('Missing ledger hostname');
+    }
+    this.ledger = ledger ||
+      new WebLedgerClient({injector, hostname, mode, logger, httpsAgent});
+
+    this.mode = mode || DEFAULT_MODE;
+    this.logger = logger || console;
+
+    this.httpsAgent = httpsAgent;
```

Comment:
We need any validation on these inputs?

Reply:
I don't think so. We can go back and add it if it becomes a problem.